nixos/modules/services/finance/taler/exchange.nix

```diff
@@ -0,0 +1,148 @@
+{
+  lib,
+  config,
+  options,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = cfgTaler.exchange;
+  cfgTaler = config.services.taler;
+
+  talerComponent = "exchange";
+
+  # https://docs.taler.net/taler-exchange-manual.html#services-users-groups-and-file-system-hierarchy
+  servicesDB = [
+    "httpd"
+    "aggregator"
+    "closer"
+    "wirewatch"
+  ];
+
+  servicesNoDB = [
+    "secmod-cs"
+    "secmod-eddsa"
+    "secmod-rsa"
+  ];
+in
+
+{
+  imports = [
+    (import ./common.nix { inherit talerComponent servicesDB servicesNoDB; })
```

Comment:
When I tried to do something similar to this in the matrix-synapse module, both @SuperSandro2000 and @Ma27 got the heebie-jeebies.

I'm not really sure of a much better solution, but I think generally people might have some apprehensions about not importing all the things through the module list.

I still run my synapse module with this, and its been fine for years, so I dont mind, but I'm calling attention to this in case anyone else has feedback here/missed something.

Reply:
@dali99 do you have a link to the PR, for more context?

Reply:
The relevant links are https://github.com/NixOS/nixpkgs/pull/242026/files#r1255517843 and https://github.com/NixOS/nixpkgs/pull/241973#pullrequestreview-1618774173

There are no specific points made about the dangers or anything, so /shrug

Reply:
One difference from here might revolve around how I passed in `cfg.package.python`. Which depends on the module system itself (and thus maybe evaluation order?). The Taler module in question here doesn't do that (passing in just normal variables), so maybe it's less of a problem.